Source/Core/DolphinQt/Config/GameConfigEdit.cpp

```diff
@@ -0,0 +1,170 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt/Config/GameConfigEdit.h"
+
+#include <QCompleter>
+#include <QDesktopServices>
+#include <QFile>
+#include <QMenu>
+#include <QTextCursor>
+#include <QTextEdit>
+#include <QVBoxLayout>
+#include <QWhatsThis>
+
+#include "DolphinQt/Config/GameConfigHighlighter.h"
+
+GameConfigEdit::GameConfigEdit(QWidget* parent, const QString& path) : m_path(path)
+{
+  LoadFile();
+
+  setAcceptRichText(false);
+  setContextMenuPolicy(Qt::CustomContextMenu);
+
+  new GameConfigHighlighter(document());
+
+  m_keyword_map[QStringLiteral("Core")] =
+      tr("Section that contains most CPU- and Hardware related settings.");
+
+  m_keyword_map[QStringLiteral("CPUThread")] = tr(
+      "Controls whether or not Dual-Core should be enabled. Can improve performance but can also "
+      "cause issues. Defaults to <b>True</b>");
+  m_keyword_map[QStringLiteral("FastDiscspeed")] =
+      tr("Shortens loading times but may break some games. Defaults to <b>False</b>");
+  m_keyword_map[QStringLiteral("Fastmem")] =
+      tr("Assumes standard SDK memory accesses. Improves speed and works for most titles.");
+  m_keyword_map[QStringLiteral("MMU")] = tr("Controls whether or not the Memory Management Unit "
+                                            "should be emulated. Required for many games.");
+  m_keyword_map[QStringLiteral("DSPHLE")] =
+      tr("Controls whether to use high- or low-level DSP emulation. Defaults to <b>True</b>");
+  m_keyword_map[QStringLiteral("JITFollowBranch")] =
+      tr("Tries to translate branches ahead of time, improving performance in most cases. Defaults "
+         "to <b>True</b>");
+
+  m_keyword_map[QStringLiteral("Gecko")] = tr("Section that contains all Gecko cheat codes.");
+  m_keyword_map[QStringLiteral("ActionReplay")] =
+      tr("Section that contains all ActionReplay cheat codes.");
+  m_keyword_map[QStringLiteral("Video_Settings")] =
+      tr("Section that contains all Graphics related settings.");
+
+  Connect();
+}
+
+void GameConfigEdit::LoadFile()
+{
+  QFile file(m_path);
+  if (!file.open(QIODevice::ReadOnly | QIODevice::Text))
+    return;
+
+  setPlainText(QString::fromStdString(file.readAll().toStdString()));
+}
+
+#include <iostream>
+
+void GameConfigEdit::SaveFile()
+{
+  if (!isVisible() || isReadOnly())
+    return;
+
+  QFile file(m_path);
+
+  if (!file.open(QIODevice::WriteOnly | QIODevice::Truncate | QIODevice::Text))
+    return;
+
+  const std::string contents = toPlainText().toStdString();
+
+  file.write(contents.c_str(), contents.length());
```

Comment:
Should this do something if the write doesn't fully succeed?

Reply:
Is that very likely? In which scenarios would that happen if ``open`` succeeded?

Reply:
Storing Dolphin on a USB drive or the settings on the USB drive (yeah, don't get me wrong it's as dumb as it sounds, but hey, users) and then removing it (or the port happening to fail after the open). It can also happen on a regular drive that happens to fail after the open, though this would likely be a much rarer case.